spec/lib/license_finder/package_managers/gradle_spec.rb

```diff
@@ -147,46 +148,138 @@ module LicenseFinder
+              expect(subject.active?).to be true
+            end
+          end
+        end
+
+        context "when there's a settings.gradle" do
+          it 'uses the build.gradle referenced inside' do
+            SETTINGS_DOT_GRADLE = <<-GRADLE
+    rootProject.buildFileName = 'build-alt.gradle'
+            GRADLE
+
+            FakeFS do
+              FileUtils.mkdir_p '/fake/path'
+              File.open('/fake/path/settings.gradle', 'w') do |file|
+                file.write SETTINGS_DOT_GRADLE
+              end
+              FileUtils.touch '/fake/path/build-alt.gradle'
+
+              expect(subject.active?).to be true
+            end
+          end
         end
       end
+    end
 
-      context "when there's no build.gradle or build.gradle.kts" do
-        it 'returns false' do
-          expect(subject.active?).to be false
+    describe '#project_root??' do
+      include FakeFS::SpecHelpers
+
+      context 'when dealing with root gradle project' do
+        context "when there's a build.gradle" do
+          it 'returns true' do
+            FakeFS do
+              FileUtils.mkdir_p '/fake/path'
+              FileUtils.touch '/fake/path/build.gradle'
+
+              expect(SharedHelpers::Cmd).to receive(:run).with("gradle properties | grep 'parent: '").and_call_original
+
+              expect(subject.project_root?).to be true
+            end
+          end
+        end
+
+        context "when there's no build.gradle or build.gradle.kts" do
+          it 'returns false' do
+            FakeFS do
+              FileUtils.mkdir_p '/fake/path'
+              expect(SharedHelpers::Cmd).not_to receive(:run).with("gradle properties | grep 'parent: '")
+              expect(subject.project_root?).to be false
+            end
+          end
+        end
+
+        context "when there's build.gradle.kts" do
+          it 'return true' do
+            FakeFS do
+              FileUtils.mkdir_p '/fake/path'
+              FileUtils.touch '/fake/path/build.gradle.kts'
+
+              expect(SharedHelpers::Cmd).to receive(:run).with("gradle properties | grep 'parent: '").and_call_original
+
+              expect(subject.project_root?).to be true
+            end
+          end
+        end
+
+        context "when there's a settings.gradle" do
+          it 'uses the build.gradle referenced inside' do
+            SETTINGS_DOT_GRADLE = <<-GRADLE
+  rootProject.buildFileName = 'build-alt.gradle'
+            GRADLE
+
+            FakeFS do
+              FileUtils.mkdir_p '/fake/path'
+              File.open('/fake/path/settings.gradle', 'w') do |file|
+                file.write SETTINGS_DOT_GRADLE
+              end
+              FileUtils.touch '/fake/path/build-alt.gradle'
+
+              expect(SharedHelpers::Cmd).to receive(:run).with("gradle properties | grep 'parent: '").and_call_original
+
+              expect(subject.project_root?).to be true
+            end
+          end
```

Comment:
It looks like this set of tests is very similar (or duplicate) of the 'active?' tests. Is there a way to used shared tests for these two cases?